statsmodels/iolib/tests/test_pickle.py

```diff
@@ -8,16 +9,26 @@
 
 
 def test_pickle():
-    tmpdir = tempfile.mkdtemp(prefix='pickle')
+    tmpdir = tempfile.mkdtemp(prefix="pickle")
     a = lrange(10)
-    save_pickle(a, tmpdir+'/res.pkl')
-    b = load_pickle(tmpdir+'/res.pkl')
+
+    # test with str
+    path_str = tmpdir + "/res.pkl"
+    save_pickle(a, path_str)
+    b = load_pickle(path_str)
+    assert_equal(a, b)
+
+    # test with pathlib
+    path_pathlib = pathlib.Path(tmpdir) / "res.pkl"
```

Comment:
Maybe use a different file name just to be sure it is writing the file. res2.pkl.